### PR TITLE
PostgreSQL -  unable to add new article

### PR DESCRIPTION
**Pull request by alikon**
#### Steps to reproduce the issue
1. Create a new article
2. edit the the last created article
3. create another new article
#### Actual result

the article is created with a strange id =0
2  not able to edit the last created one
3  not able to save cause of duplicate key
#### Expected result

you can edit & create article
#### Addictional Comments

 pk value is passed like a string sometimes i'm afraid

### CODE REVIEW

File: libraries/joomla/database/driver/postgresql.php [hunk 1]
```diff
@@ -1108,7 +1108,7 @@ public function insertObject($table, &$object, $key = null)
 			}
 
 			// Ignore any internal fields or primary keys with value 0.
-			if (($k[0] == "_") || ($k == $key && $v === 0))
+			if (($k[0] == "_") || ($k == $key && (($v === 0)||($v === '0'))))
```
Comment (zero-24) [inline libraries/joomla/database/driver/postgresql.php]: can we have a space between the closing `)`?

so we get this: `if (($k[0] == "_") || ($k == $key && (($v === 0) || ($v === '0'))))`

**Comment by alikon** [comment]
made it

**Comment by zero-24** [comment]
:+1:

**Comment by waader** [comment]
I have tested this item :white_check_mark: successfully on ffcdd802c5082b5e52fc47f220cb315ca3a8f612<br /><br />:+1:<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8358">issues.joomla.org/joomla-cms/8358</a>.</sub>

**Comment by wmgraf** [comment]
I have tested this item :white_check_mark: successfully on ffcdd802c5082b5e52fc47f220cb315ca3a8f612<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8358">issues.joomla.org/joomla-cms/8358</a>.</sub>

**Comment by zero-24** [comment]
RTC. Thanks.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8358">issues.joomla.org/joomla-cms/8358</a>.</sub>

**Pull request merged**